### DISCUSSION TITLE
Consider adding a Weaviate Provider for Apache Airflow

### DISCUSSION THREAD

**Discussion by MarcusSorealheis** [apache/airflow#29186] [Ideas] [answered]
I started a [branch ](https://github.com/MarcusSorealheis/incubator-airflow/tree/add-weaviate-provider) possibly in vain because I love Weaviate's open source vector search capabilities and understand the benefits of DAGs for these types of workloads. I welcome a discussion about what it would take to get this done and if people would be open to such a request. 

As a slightly related aside, due to some accessibility compliance of a previous employer, I jumped in and made an extremely modest yet important contribution to Airflow. IOW, I've been here before but it's been years. My fork is still incubator.

**Comment by Taragolis** [comment, marked as answer]
@MarcusSorealheis I would recommend discuss this first in [devlist](https://airflow.apache.org/community/) first. AFAIK there are quite a few requirements for create new community supported provider this include but not limited by include System Tests (see: [AIP-47](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests))

However providers could maintained outside of community providers (in-house) and can be added as part of [Airflow Ecosystem](https://airflow.apache.org/ecosystem/#third-party-airflow-plugins-and-providers)

**Comment by MarcusSorealheis** [comment]
Somebody at Astronomer agreed! 

https://docs.astronomer.io/learn/airflow-weaviate

Congratulations and good work.
